src/test/scala/strawman/collection/test/TraverseTest.scala

```diff
@@ -3,51 +3,95 @@ package strawman
 package collection.test
 
 import org.junit.Test
-import strawman.collection.Iterable
-import strawman.collection.mutable.{ArrayBuffer, Builder, Growable}
 import strawman.collection._
+import strawman.collection.mutable.{ArrayBuffer, Builder, Growable}
 
 import scala.{Any, Either, Int, Left, None, Option, Right, Some, Unit}
+import java.lang.String
 import scala.Predef.ArrowAssoc
 import scala.math.Ordering
-import java.lang.String
 
 class TraverseTest {
 
-  def optionSequence[CC[X] <: Iterable[X] with IterableOps[X, CC, _], A](xs: CC[Option[A]]): Option[CC[A]] = {
-    def folder[F[X] <: Growable[X]]: (Option[F[A]], Option[A]) => Option[F[A]] = { (bo, xo) =>
-      (bo, xo) match {
-        case (Some(builder), Some(a)) => Some(builder += a)
-        case _ => None
-      }
-    }
-    val factory = xs.iterableFactory
-    factory match {
-      case iterableBuilder: IterableFactoryWithBuilder[CC] =>
-        xs.foldLeft[Option[Builder[A, CC[A]]]](
-          Some(iterableBuilder.newBuilder[A]())
-        )(
-          folder[({ type l[X] = Builder[X, CC[X]] })#l]
-        ).map(_.result)
-      case _ =>
-        xs.foldLeft[Option[ArrayBuffer[A]]](Some(new ArrayBuffer[A]))(folder).map(_.to(xs.iterableFactory))
-    }
-  }
+  // You can either overload methods for IterableOps and Iterable with SortedOps (if you want to support constrained collection types)
+  def optionSequence1[C[X] <: IterableOps[X, C, _], A](xs: C[Option[A]]): Option[C[A]] =
+    xs.foldLeft[Option[Builder[A, C[A]]]](Some(Builder.from[A, C](xs.iterableFactory))) {
+      case (Some(builder), Some(a)) => Some(builder += a)
+      case _ => None
+    }.map(_.result)
+  def optionSequence1[C[X] <: Iterable[X] with SortedOps[X, C[X], C], A : Ordering](xs: C[Option[A]]): Option[C[A]] =
+    xs.foldLeft[Option[Builder[A, C[A]]]](Some(Builder.from[A, C](xs.sortedIterableFactory))) {
+      case (Some(builder), Some(a)) => Some(builder += a)
+      case _ => None
+    }.map(_.result)
+
+  // ...or use BuildFrom to abstract over both and also allow building arbitrary collection types
+  def optionSequence2[CC[X] <: Iterable[X], A, To](xs: CC[Option[A]])(implicit bf: BuildFrom[CC[Option[A]], A, To]): Option[To] =
```

Comment:
This should work for the tests below if you ask for a `StrictBuildFrom` instead of a `BuildFrom` but it doesn't because `Set` is not `Buildable` at the moment. I'm currently working on changing that but I ran into some complications, so I wanted to get this PR out first. Making Sets buildable is a separate concern.